plugin-tsv/src/main/java/ca/on/oicr/gsi/shesmu/tsv/StringExpandFile.java

```diff
@@ -0,0 +1,80 @@
+package ca.on.oicr.gsi.shesmu.tsv;
+
+import ca.on.oicr.gsi.shesmu.plugin.Definer;
+import ca.on.oicr.gsi.shesmu.plugin.PluginFile;
+import ca.on.oicr.gsi.shesmu.plugin.functions.ShesmuMethod;
+import ca.on.oicr.gsi.shesmu.plugin.functions.ShesmuParameter;
+import ca.on.oicr.gsi.status.SectionRenderer;
+import io.prometheus.client.Gauge;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.xml.stream.XMLStreamException;
+
+class StringExpandFile extends PluginFile {
+
+  public static final Set<String> tailList(String[] items) {
+    final Set<String> result = new TreeSet<>();
+    for (int i = 1; i < items.length; i++) {
+      result.add(items[i]);
+    }
+    return result;
+  }
+
+  private static final Pattern TAB = Pattern.compile("\t");
+  private static final Gauge tableBad =
+      Gauge.build("shesmu_strexpand_lookup_bad", "A string expansion table is badly formed.")
+          .labelNames("fileName")
+          .register();
+  private final Definer definer;
+  private Map<String, Set<String>> expansions = Collections.emptyMap();
+  private boolean good;
+
+  public StringExpandFile(Path fileName, String instanceName, Definer<StringExpandFile> definer) {
+    super(fileName, instanceName);
+    this.definer = definer;
+  }
+
+  @ShesmuMethod(
+      type = "as",
+      description =
+          "Expand a string to a set of strings or a set of containing only the input string.")
+  public Set<String> $(
```

Comment:
Why the dollar sign as the variable name?

Reply:
The `$` gets substituted to the main part of the file name. I've given the method a more sensible name and moved this name into the annotations.